dags/process_elife_zip_dag.py

```diff
@@ -141,22 +145,44 @@ def convert_tiff_images_in_expanded_bucket_to_jpeg_images(**context) -> None:
 def update_tiff_references_to_jpeg_in_article(**context) -> bytes:
     zip_file_name = get_file_name_passed_to_dag_run_conf_file(context)
     article_name = get_expected_elife_article_name(zip_file_name)
-    folder_name = zip_file_name.replace('.zip', '/')
-    s3_key = folder_name + article_name
+    prefix = zip_file_name.replace('.zip', '/')
+    s3_key = prefix + article_name
     s3 = get_s3_client()
     response = s3.get_object(Bucket=DESTINATION_BUCKET, Key=s3_key)
     article_bytes = BytesIO(response['Body'].read())
     article_xml = etree.parse(article_bytes)
     matches = article_xml.xpath('//*[@mimetype="image" and @mime-subtype="tiff"]')
     if matches:
         for element in matches:
-            href = '{http://www.w3.org/1999/xlink}href'
-            element.attrib[href] = re.sub(r'\.\w+$', '.jpg', element.attrib[href])
+            element.attrib[XLINK_HREF] = re.sub(r'\.\w+$', '.jpg', element.attrib[XLINK_HREF])
             element.attrib['mime-subtype'] = 'jpeg'
 
     return etree.tostring(article_xml, xml_declaration=True, encoding='UTF-8')
 
 
+def add_missing_jpeg_extensions_in_article(**context) -> bytes:
+    zip_file_name = get_file_name_passed_to_dag_run_conf_file(context)
+    article_name = get_expected_elife_article_name(zip_file_name)
+    prefix = zip_file_name.replace('.zip', '/')
+    s3_key = prefix + article_name
+    s3 = get_s3_client()
+    response = s3.get_object(Bucket=DESTINATION_BUCKET, Key=s3_key)
+    article_bytes = BytesIO(response['Body'].read())
+    article_xml = etree.parse(article_bytes)
+    matches = article_xml.xpath(
+        '//*[@mimetype="image" and @mime-subtype="jpeg"]')
+    if matches:
```

Comment:
what's the negative value of `matches`? If it's something like `[]` only the `for` (on an empty list) is needed rather than also the `if`?

Reply:
removed. not sure what I was thinking.